client/me/account/account-email-fragment.tsx

```diff
@@ -0,0 +1,242 @@
+		noticeText = translate( '%(email)s is not a valid email address.', {
+			args: {
+				email: getUserSetting( {
+					settingName: 'user_email',
+					unsavedUserSettings,
+					userSettings,
+				} ),
+			},
+		} );
+	}
+
+	return <FormInputValidation isError={ true } text={ noticeText } />;
+};
+
+const AccountEmailPendingEmailChangeNotice = ( {
+	unsavedUserSettings,
+	userSettings,
+}: {
+	unsavedUserSettings: UserSettingsType;
+	userSettings: UserSettingsType;
+} ): JSX.Element | null => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const domainsList: ResponseDomain[] = useSelector( getFlatDomainsList );
+	const isRequestingDomainList = useSelector( isRequestingAllDomains );
+	const isEmailChangePending = useSelector( isPendingEmailChange );
+
+	if ( ! isEmailChangePending ) {
+		return null;
+	}
+
+	const editContactInfoInBulkUrl = `/domains/manage?site=all&action=edit-contact-email`;
+	const email = getUserSetting( {
+		settingName: 'new_user_email',
+		unsavedUserSettings,
+		userSettings,
+	} );
+
+	const hasCustomDomainRegistration = domainsList.some( ( domain ) => {
+		return domainTypes.REGISTERED === domain.type;
+	} );
+
+	const noticeText =
+		isRequestingDomainList || ! hasCustomDomainRegistration
+			? translate(
+					'Your email change is pending. Please take a moment to check %(email)s for an email with the subject "[WordPress.com] New Email Address" to confirm your change.',
+					{
+						args: { email },
+					}
+			  )
+			: translate(
+					'Your email change is pending. Please take a moment to:{{br/}}1. Check %(email)s for an email with the subject "[WordPress.com] New Email Address" to confirm your change.{{br/}}2. Update contact information on your domain names if necessary {{link}}here{{/link}}.',
+					{
+						args: {
+							email,
+						},
+						components: {
+							br: <br />,
+							link: <a href={ editContactInfoInBulkUrl } />,
+						},
+					}
+			  );
+
+	return (
+		<Notice showDismiss={ false } status="is-info" text={ noticeText }>
+			<NoticeAction onClick={ () => dispatch( cancelPendingEmailChange() ) }>
+				{ translate( 'Cancel' ) }
+			</NoticeAction>
+		</Notice>
+	);
+};
+
+const AccountEmailFragment = ( {
+	emailInputId = 'user_email',
+	emailInputName = 'user_email',
+	emailValidationHandler,
+	getFocusHandler = noopFocusHandler,
+	isEmailControlDisabled = false,
+	unsavedUserSettings = {},
+	userSettings = {},
+}: AccountEmailFragmentProps ): JSX.Element => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const isEmailChangePending = useSelector( isPendingEmailChange );
+
+	const [ emailInvalidReason, setEmailInvalidReason ] = useState< AccountEmailValidationReason >(
+		EMAIL_VALIDATION_REASON_IS_VALID
+	);
+
+	const emailAddress = isEmailChangePending
+		? getUserSetting( { settingName: 'new_user_email', unsavedUserSettings, userSettings } )
+		: getUserSetting( { settingName: 'user_email', unsavedUserSettings, userSettings } );
```

Comment:
Here is another approach we could consider:

```suggestion
	const emailAddress = getUserSetting( {
		settingName: isEmailChangePending ? 'new_user_email' : 'user_email',
		unsavedUserSettings,
		userSettings
	} );
```

Reply:
This is much cleaner! 😍